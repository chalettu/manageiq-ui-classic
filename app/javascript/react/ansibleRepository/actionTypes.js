const prefix = '@@manageiq/ansibleCatalogItem/';
export const LOAD_CREDENTIALS = `${prefix} LOAD_CREDENTIALS`;
export const LOAD_REPO = `${prefix} LOAD_REPO`;
export const LOAD_MANAGER_RESOURCE = `${prefix} LOAD_MANAGER_RESOURCE`;
