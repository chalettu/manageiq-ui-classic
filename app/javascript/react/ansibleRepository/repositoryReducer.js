import * as actionTypes from './actionTypes';

const initialState = {
  credentials: [],
  repo: {}
};

export const repositoryReducer = (state = initialState, action) => {
  const updatedState = {...state};
  switch (action.type) {
    case actionTypes.LOAD_CREDENTIALS:
    updatedState.credentials = action.payload.map(credential => ({label: credential.name, value: credential.id}));
    break;
    case actionTypes.LOAD_MANAGER_RESOURCE:
    updatedState.managerResource = action.payload;
    break;
    case actionTypes.LOAD_REPO:
    updatedState.repo = action.payload;
    default:
    return updatedState;
  }
  return updatedState;
};
