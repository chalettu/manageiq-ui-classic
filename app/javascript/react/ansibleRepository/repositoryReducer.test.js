import {repositoryReducer} from './repositoryReducer';
import * as actionTypes from './actionTypes';
import { testReducerSnapshotWithFixtures } from '../../components/test-utils';
import * as testFixtures from './test.fixtures';

const resources = testFixtures.testRecords;
const fixtures = {
  'should return the initial state': {},
  'should load a credential': {
    action: {
      type: actionTypes.LOAD_CREDENTIALS,
      payload:  testFixtures.testCredentialRecords,
    },
  },
  'should load a list of Manager resources': {
    action: {
      type: actionTypes.LOAD_MANAGER_RESOURCE,
      payload: { resources: testFixtures.managerResource },
    },
  },
  'should load a repo': {
    action: {
      type: actionTypes.LOAD_REPO,
      payload: testFixtures.existingRepo,
    },
  },
};
describe('Ansible repository reducer', () => testReducerSnapshotWithFixtures(repositoryReducer, fixtures));
