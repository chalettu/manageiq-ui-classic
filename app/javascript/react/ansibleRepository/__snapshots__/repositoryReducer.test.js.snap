// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Ansible repository reducer should load a credential 1`] = `
Object {
  "credentials": Array [
    Object {
      "label": "test",
      "value": 1,
    },
  ],
  "repo": Object {},
}
`;

exports[`Ansible repository reducer should load a list of Manager resources 1`] = `
Object {
  "credentials": Array [],
  "managerResource": Object {
    "resources": Array [
      Object {
        "href": "http://test.com",
      },
    ],
  },
  "repo": Object {},
}
`;

exports[`Ansible repository reducer should load a repo 1`] = `
Object {
  "credentials": Array [],
  "repo": Object {
    "authentication_id": null,
    "description": "test",
    "name": "Test",
    "scm_branch": "master",
    "scm_clean": false,
    "scm_delete_on_update": false,
    "scm_type": "git",
    "scm_update_on_launch": false,
    "scm_url": "https://github.com/test",
  },
}
`;

exports[`Ansible repository reducer should return the initial state 1`] = `
Object {
  "credentials": Array [],
  "repo": Object {},
}
`;
